src/modm/platform/usb/sam/usb.hpp

```diff
@@ -32,17 +32,20 @@ class Usb
 		NVIC_SetPriority(USB_IRQn, priority);
 	}
 
-	template< template<Peripheral _> class... Signals >
+	template< class... Pins >
 	static void
 	connect()
 	{
-		using Connector = GpioConnector<Peripheral::Usb, Signals...>;
-		using Dp = typename Connector::template GetSignal< Gpio::Signal::Dp >;
-		using Dm = typename Connector::template GetSignal< Gpio::Signal::Dm >;
-		static_assert(((Connector::template IsValid<Dp> and Connector::template IsValid<Dm>) and sizeof...(Signals) == 2),
-					  "Usb::connect() requires one Dp and one Dm signal!");
-
-		Connector::connect();
+		using DpPin = GetPin_t<PeripheralPin::Dp, Pins...>;
+		using DmPin = GetPin_t<PeripheralPin::Dm, Pins...>;
+		static_assert(
+			!std::is_same_v<typename DpPin::Base, typename DmPin::Base>,
+			"Dp and Dm cannot use the same pin!");
```

Comment:
Why not use the trick from above here too?
```suggestion
		template<class DpPin, class DmPin>
		struct ValidateNotSame {
			static_assert(
				!std::is_same_v<typename DpPin::Base, typename DmPin::Base>,
				"Dp and Dm cannot use the same pin!");
		};
```

Reply:
We require `DpPin` to use the `Dp` peripheral pin and `DmPin` to use `Dm`, so it can't happen that the pins use the same peripheral pin.

This is a problem on SAM UART because there the same signal (e.g. `Pad0`) can be used as both `Rx` and `Tx` peripheral pins.